binary_and_decimal.rb

```diff
@@ -1,83 +1,77 @@
 # 1. Write a method named `binary_to_decimal` that creates an array of size 8.
-#    Fill the array randomly with 0’s and 1’s. Print out the array so that it
+#    Fill the array randomly with 0's and 1's. Print out the array so that it
 #    appears as a binary number. Calculate the decimal value for this binary
-#    number using the algorithm you devised in class. Print out the decimal value.
+#    number using the algorithm you devised in class. Print out the decimal val.
 def binary_to_decimal
   # binary = Array.new(8)
-  binary = Array.new(8).map {|n| rand(2)}
+  binary = Array.new(8).map { rand(2) }
   decimal = 0
 
   binary.reverse.each_with_index do |digit, i|
-    if digit == 1
-      decimal += 2 ** i
-    end
+    decimal += 2**i if digit == 1
   end
-
-  return decimal
-
+  decimal
 end
```

Comment:
The logic looks good! The problem statement asks you to print both the binary and decimal values. You are just missing the puts statement in the method.